Unable to compile on Linux

Attempting to build on my laptop running Arch Linux I get the following error. Seems something is off with the conditional compilation.

Rust version: rustc 1.18.0 (03fc9d622 2017-06-06)

Error:

```
error[E0201]: duplicate definitions with name `mode`:
  --> /home/wmoore/.cargo/registry/src/github.com-1ecc6299db9ec823/fe-1.0.3/src/fileinfo.rs:82:5
   |
76 | /     fn mode(metadata: &fs::Metadata) -> mode_t {
77 | |         use std::os::linux::fs::MetadataExt;
78 | |         metadata.st_mode()
79 | |     }
   | |_____- previous definition of `mode` here
...
82 | /     fn mode(metadata: &fs::Metadata) -> mode_t {
83 | |         use std::os::unix::fs::MetadataExt;
84 | |         metadata.mode()
85 | |     }
   | |_____^ duplicate definition

error: aborting due to previous error

error: failed to compile `fe v1.0.3`, intermediate artifacts can be found at `/tmp/cargo-install.g7q15W4XZOWZ`
```

--- a/src/fileinfo.rs
+++ b/src/fileinfo.rs
@@ -78,7 +78,7 @@ impl FileInfo {
         metadata.st_mode()
     }
 
-    #[cfg(all(not(target_os = "macos"), unix))]
+    #[cfg(all(not(target_os = "macos"), not(target_os = "linux"), unix))]
     fn mode(metadata: &fs::Metadata) -> mode_t {
         use std::os::unix::fs::MetadataExt;
         metadata.mode()

